Adding test to cover default serialization behavior

ealsur opened the pull request:
# Default serialization test

## Description

This PR adds a test to assert the default serialization behavior (`NullValueHandling = NullValueHandling.Ignore`) and make sure we don't regress it in any future change.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Closing issues

This PR closes #350

simplynaveen20 commented:
While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization  allow null if user is setting it ?

simplynaveen20 commented on Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosJsonSerializerTests.cs:
```diff
@@ -64,6 +64,30 @@ public async Task TestCustomJsonSerializer()
+            StreamReader reader = new StreamReader(cosmosResponseMessage.Content);
+            string text = reader.ReadToEnd();
+
+            Assert.AreEqual(-1, text.IndexOf("description"), "Stored document contains a null attribute");
```
"description" -> nameof(document.description)

ealsur replied:
Done

ealsur commented:
> While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization allow null if user is setting it ?

If we change that behavior, we just adapt the test. The goal of the test is not to do any accidental regressions, but a conscious one.

simplynaveen20 commented:
> > While working on the LINQ tests , existing V2 test were failing which were expecting null object , I am working on item where i changed this behavior and removed NullValueHandling.Ignore from the default serializer . Shouldn't the default of serialization allow null if user is setting it ?
> 
> If we change that behavior, we just adapt the test. The goal of the test is not to do any accidental regressions, but a conscious one.

Make sense , please go forward with this , and if i change the behavior in my work item  i will juts flip the assert

kirankumarkolli commented:
Thanks @ealsur
